code/core/src/main/java/com/adobe/marketing/mobile/rulesengine/LogicalExpression.java

```diff
@@ -26,52 +25,50 @@ public LogicalExpression(final List<Evaluable> operands, final String operationN
 
     @Override
     public RulesResult evaluate(final Context context) {
-        ArrayList<RulesResult> resolvedOperands = new ArrayList<>();
 
         if (operationName == null || operationName.isEmpty()) {
             return new RulesResult(
                     RulesResult.FailureType.MISSING_OPERATOR,
                     "Null or empty operator for logical expression");
         }
-
-        if (operands != null) {
-            for (Evaluable evaluable : operands) {
-                if (evaluable != null) {
-                    resolvedOperands.add(evaluable.evaluate(context));
-                }
-            }
-        }
-
         switch (operationName) {
             case "and":
-                return performAndOperation(resolvedOperands);
+                return performAndOperation(context, operands);
             case "or":
-                return performOrOperation(resolvedOperands);
+                return performOrOperation(context, operands);
             default:
                 return new RulesResult(
                         RulesResult.FailureType.MISSING_OPERATOR,
                         String.format("Unknown conjunction operator - %s.", operationName));
         }
     }
 
-    private RulesResult performAndOperation(final List<RulesResult> resolvedOperands) {
-        for (RulesResult rulesResult : resolvedOperands) {
-            if (!rulesResult.isSuccess()) {
-                return new RulesResult(
-                        RulesResult.FailureType.CONDITION_FAILED, "AND operation returned false.");
+    private RulesResult performAndOperation(
+            final Context context, final List<Evaluable> resolvedOperands) {
```

Comment:
We are passing the application context we are passing here right?
Just want to make sure (for my clarity)

Reply:
Haha, it's not the Android application context, it's the SDK context object used by Rules Engine. https://github.com/adobe/aepsdk-core-android/blob/main/code/core/src/main/java/com/adobe/marketing/mobile/rulesengine/Context.java